node/src/components/linear_chain_sync/error.rs

```diff
@@ -47,7 +96,4 @@ where
 
     #[error(transparent)]
     TrieFetcherError(#[from] FetcherError<Trie<Key, StoredValue>, I>),
-
-    #[error("Could not store block header: {block_header}")]
-    CouldNotStoreBlockHeader { block_header: Box<BlockHeader> },
```

Comment:
@afck This was being thrown when the storage component was reporting the block header was already present.  But this _could_ happen on main-net, especially if the user joins with the same trusted hash over and over again.  So I am removing this error.

Reply:
Is the header being already present the only thing that can go wrong in `put_block_header_to_storage`? Otherwise, instead of a `bool`, this should probably return a `Result`?

Reply:
It _should_, but storage currently swallows errors. It's worth revisiting that architecture, but it's beyond the scope of this PR.